### PR TITLE
advancedTLS: Documentation

### PR DESCRIPTION
**Pull request by gtcooke94**
Add documentation to the main package, the `RootCertificateOptions` struct, and the `IdentityCertificateOptions` struct.

RELEASE NOTES: none

### CODE REVIEW

File: security/advancedtls/advancedtls.go [hunk 1]
```diff
@@ -16,9 +16,14 @@
  *
  */
 
-// Package advancedtls is a utility library containing functions to construct
-// credentials.TransportCredentials that can perform credential reloading and
-// custom verification check.
+// Package advancedtls is a library containing APIs for configuring grpc
```
Comment (dfawley) [inline security/advancedtls/advancedtls.go:19]: Nit: all packages are libraries, and all libraries contain APIs, so I think you could remove some words here to make everything more to-the-point.  How about something like:

`Package advancedtls provides gRPC transport credentials that allow easy configuration of advanced TLS features.`

File: security/advancedtls/advancedtls.go [hunk 2]
```diff
@@ -119,8 +124,9 @@ type GetRootCAsResults = RootCertificates
 
 // RootCertificateOptions contains options to obtain root trust certificates
 // for both the client and the server.
-// At most one option could be set. If none of them are set, we
-// use the system default trust certificates.
+// At most one option should be set. If none of them are set, we use the system
```
Comment (dfawley) [inline security/advancedtls/advancedtls.go:127]: What is an "option"?  Is each option a single field in this struct?  If so, maybe "at most one field should be set"?

File: security/advancedtls/advancedtls.go [hunk 3]
```diff
@@ -153,18 +159,18 @@ func (o RootCertificateOptions) nonNilFieldCount() int {
 
 // IdentityCertificateOptions contains options to obtain identity certificates
 // for both the client and the server.
-// At most one option could be set.
```
Comment (dfawley) [inline security/advancedtls/advancedtls.go:156]: As above.

File: security/advancedtls/advancedtls.go (hunk 1 above)
Comment (gtcooke94) [reply security/advancedtls/advancedtls.go]: Changed, I like that

File: security/advancedtls/advancedtls.go (hunk 2 above)
Comment (gtcooke94) [reply security/advancedtls/advancedtls.go]: Changed, good call

File: security/advancedtls/advancedtls.go (hunk 3 above)
Comment (gtcooke94) [reply security/advancedtls/advancedtls.go:156]: Changed

**Review by dfawley — APPROVED** [review]

**Pull request merged**